src/util/db/sqlqueryfinisher.h

```diff
@@ -0,0 +1,38 @@
+#ifndef MIXXX_SQLQUERYFINISHER_H
+#define MIXXX_SQLQUERYFINISHER_H
+
+
+#include <QSqlQuery>
+
+
+// Ensures that reusable queries are properly finished when
+// leaving the corresponding execution scope. This will free
+// resources of prepared statements until the next execution.
+class SqlQueryFinisher final {
+public:
+    explicit SqlQueryFinisher(QSqlQuery query)
+        : m_query(query) {
+    }
+    SqlQueryFinisher(SqlQueryFinisher&& other)
+        : m_query(std::move(other.m_query)) {
```

Comment:
The compiler is only able to favour "move" over "copy" if I give hint by using std::move(). The same is true for all other move constructors where some Qt members are implicitly shared. Implicit sharing is somwhere between "copy" and "move". But maybe sometime in the future they will provide an efficient move constructor? By using std::move() now we are prepared if this ever happens.